site/content/docs/gsoc-2020-ideas.md

```diff
@@ -94,14 +101,30 @@ friendly front end could be created. This would allow a designer to select the
 signals they want to capture and the necessary logic analyser would be
 generated with all the required wiring to feed to signals into the analyser.
 
-## Implement the bit manipulation extension for Ibex
+Mentors: Greg, Pirmin
+
+## Proof-of-concept integration of pointer authentication support in Ibex
+
+**Summary:** Extend the processor pipeline of [Ibex]
+(https://github.com/lowrisc/ibex) with instructions to generate and
+check pointer authentication codes.
+
+[Pointer Authentication]
+(https://www.usenix.org/system/files/sec19fall_liljestrand_prepub.pdf) uses
+cryptographic message authentication codes (MACs) both generated and
+authenticated at runtime to protect the integrity of
+pointers in order to aggravate attacks targeting arbitrary code execution
+through malicious manipulation of code and data pointers. This project aims
+at doing a proof-of-concept integration of pointer authentication
+into Ibex.
 
-**Summary:** Implement support for the proposed [RISC-V bit manipulation
-extension](https://github.com/riscv/riscv-bitmanip) in
-[Ibex](https://github.com/lowrisc/ibex).
+For this project, the actual MAC is considered a black box of configurable
+latency. The focus lies on integrating pointer authentication into the
+processor pipeline. To this end, new custom instructions need to be added to
+generate and authenticate pointers based on the value of the PC, SP and a
+secret key (including exception handling if authentication fails).
 
-As well as the RTL implementation, an important part of this project will be
-to deliver a workable testing strategy.
+Mentors: Pirmin
```

Comment:
If you want a backup mentor on this, I'd be happy to help from the LLVM side.

Reply:
Sounds great, thanks @lenary !

Reply:
I think @luismarques is also interested.

Reply:
Alright, I'll add @luismarques , too.